module/registry.mjs

```diff
@@ -521,63 +521,64 @@ class SummonRegistry {
 }
 
 /* -------------------------------------------- */
-/*  Public API                                  */
+/*  Ready API                                   */
 /* -------------------------------------------- */
 
 /**
  * Track the ready status of various registries.
  * @type {Map<string, boolean>}
  */
 const RegistryStatus = new class extends Map {
+  constructor(iterable) {
+    super(iterable);
+    const { promise, resolve } = Promise.withResolvers();
```

Comment:
Love `withResolvers`! (and I tested it with the app, so we should be good to use it despite it being fairly new)